planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp

```diff
@@ -840,50 +833,39 @@ bool PullOverModule::hasFinishedPullOver()
-    lanelet::utils::getArcCoordinates(status_.pull_over_lanes, modified_goal_pose_);
-  const auto arc_position_current_pose =
-    lanelet::utils::getArcCoordinates(status_.pull_over_lanes, planner_data_->self_pose->pose);
-  const double distance_to_goal = arc_position_goal_pose.length - arc_position_current_pose.length;
-
-  const double velocity = std::abs(planner_data_->self_odometry->twist.twist.linear.x);
-  if (
-    (distance_to_goal < parameters_.hazard_on_threshold_distance &&
-     velocity < parameters_.hazard_on_threshold_velocity) ||
-    status_.planner->getPlannerType() == PullOverPlannerType::ARC_BACKWARD) {
-    hazard_signal.command = HazardLightsCommand::ENABLE;
-    const double distance_from_front_to_goal =
-      distance_to_goal - planner_data_->parameters.base_link2front;
-    return std::make_pair(hazard_signal, distance_from_front_to_goal);
-  }
+  TurnSignalInfo turn_signal{};  // output
 
-  return std::make_pair(hazard_signal, std::numeric_limits<double>::max());
-}
-
-std::pair<TurnIndicatorsCommand, double> PullOverModule::getTurnInfo() const
-{
-  std::pair<TurnIndicatorsCommand, double> turn_info{};
-
-  const double distance_from_vehicle_front = std::invoke([&]() {
-    const auto arc_position_current_pose =
-      lanelet::utils::getArcCoordinates(status_.current_lanes, planner_data_->self_pose->pose);
-    const auto arc_position_end_pose =
-      lanelet::utils::getArcCoordinates(status_.current_lanes, status_.pull_over_path.end_pose);
-    return arc_position_end_pose.length - arc_position_current_pose.length -
-           planner_data_->parameters.base_link2front;
-  });
+  const auto & current_pose = planner_data_->self_pose->pose;
+  const auto & start_pose = status_.pull_over_path.start_pose;
+  const auto & end_pose = status_.pull_over_path.end_pose;
+  const auto & full_path = getFullPath();
 
-  TurnIndicatorsCommand turn_signal{};
-  const bool is_before_parking_end = distance_from_vehicle_front >= 0.0;
-  turn_signal.command =
-    is_before_parking_end ? TurnIndicatorsCommand::ENABLE_LEFT : TurnIndicatorsCommand::NO_COMMAND;
-  turn_info.first = turn_signal;
-  turn_info.second = distance_from_vehicle_front;
-  return turn_info;
+  // calc TurnIndicatorsCommand
+  {
+    const double distance_to_end =
+      calcSignedArcLength(full_path.points, current_pose.position, end_pose.position);
+    const bool is_before_end_pose = distance_to_end >= 0.0;
+    turn_signal.turn_signal.command =
+      is_before_end_pose ? TurnIndicatorsCommand::ENABLE_LEFT : TurnIndicatorsCommand::NO_COMMAND;
+  }
+
+  // calc desired/required start/end point
+  {
+    // ego decelerates so that current pose is the point `turn_light_on_threshold_time` seconds
+    // before starting pull_over
+    const double distance_to_start =
+      calcSignedArcLength(full_path.points, current_pose.position, start_pose.position);
+    const bool is_before_start_pose = distance_to_start >= 0.0;
+    turn_signal.desired_start_point =
```

Comment:
desired start point = 3 seconds before the start point of pullover

Reply:
fixed in https://github.com/tier4/autoware.universe/pull/138/commits/f6663c589953bb7cc5366ada9ba30f6833a2a772